### PR TITLE
[release/v2.8] Move from Drone to GitHub Actions

### PR DESCRIPTION
**Pull request by tashima42**
## Problem

Drone is not being actively developed anymore.
 
## Solution

Migrate the CI to GitHub Actions
**Note:** This PR is does not include tests, they are being worked on another branch.
 
### Manual Testing

Forked rancher and executed the workflows in hosted GHA runners.

### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

Existing / newly added automated tests that provide evidence there are no regressions:

### CODE REVIEW

**Review by briandowns — DISMISSED** [review]

File: .github/workflows/pull-request.yml
Comment (HarrisonWAffel) [inline .github/workflows/pull-request.yml:82]: Are we pushing images to dockerhub every time a PR is opened? Right now we only push when a merge happens, right?

File: .github/workflows/pull-request.yml
Comment (HarrisonWAffel) [inline .github/workflows/pull-request.yml:38]: It looks like we're not running any of the unit / integration tests - is that going to be added in the near future?

File: package/Dockerfile
Comment (HarrisonWAffel) [inline package/Dockerfile:33]: Why get rid of the default value?

File: scripts/version
Comment (HarrisonWAffel) [inline scripts/version:64]: Don't we need to keep this conditional (But with `GIT_TAG` instead of `DRONE_TAG`)? 

If I'm reading this correctly, `TAG` (and thus `AGENT_TAG`) will always be `${VERSION}${SUFFIX}`. Maybe I'm missing another area where `TAG` is set before this line?

**Review by HarrisonWAffel — CHANGES_REQUESTED** [review]
Couple of comments, let me know if I've missed something obvious wrt unit / integration tests

File: .github/workflows/pull-request.yml
Comment (HarrisonWAffel) [reply .github/workflows/pull-request.yml:82]: Never mind, realized this doesn't push anything to dockerhub and is being used to build locally

File: .github/workflows/pull-request.yml
Comment (tashima42) [reply .github/workflows/pull-request.yml:38]: Unit tests are in the `validate.yml` file, but integration tests are not ready yet, they will be done in another PR.

File: package/Dockerfile
Comment (tashima42) [reply package/Dockerfile]: No particular reason, I'm adding it back.

File: scripts/version
Comment (tashima42) [reply scripts/version:64]: TAG is being set at the workflow level, because it is used in other places than the script. I'll add it back with the correct env names.

File: .github/workflows/pull-request.yml
Comment (HarrisonWAffel) [reply .github/workflows/pull-request.yml:38]: From what I can see `validate.yaml` just runs linters and go mod checks. Unit tests are run in the `scripts/test` file, [here](https://github.com/rancher/rancher/blob/71c10185d52f3e1dc30b941317c71805133e6a17/scripts/test#L25-L27), just before we run the integration tests. I'm not sure we should merge this without also running the tests, but I understand getting the integration tests running in gha is no small feat.

Comment (tashima42) [reply .github/workflows/pull-request.yml:38]: You're right, running `scripts/test` is being done in a different branch

File: .github/workflows/push-release.yml
Comment (HarrisonWAffel) [inline .github/workflows/push-release.yml:168]: This step seems to be empty, from looking at the name of the step should the if statement in the above step be run here instead?

File: .github/workflows/validate.yml
Comment (pjbgf) [inline .github/workflows/validate.yml:23]:
```suggestion
        run: ./scripts/validate

```

File: .github/workflows/push-release.yml
Comment (macedogm) [inline .github/workflows/push-release.yml:54]: Where is this permission needed?

File: package/Dockerfile
Comment (macedogm) [inline package/Dockerfile:1]:
```suggestion
FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.21 as build
```

Comment (macedogm) [reply package/Dockerfile:1]: To align with https://github.com/rancher/rancher/pull/45674/files#diff-a5c0f08bc43f53a0451cad4a4a2e435ce55ad34183350dfae5e4b2c7dca6360bR14

File: package/Dockerfile.agent
Comment (macedogm) [inline package/Dockerfile.agent:6]:
```suggestion
FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.21 as build
```

Comment (macedogm) [reply package/Dockerfile.agent:6]: To align with https://github.com/rancher/rancher/pull/45674/files#diff-a5c0f08bc43f53a0451cad4a4a2e435ce55ad34183350dfae5e4b2c7dca6360bR14

**Review by macedogm — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yml
Comment (tashima42) [reply .github/workflows/pull-request.yml:137]: This is intended, since the agent image uses the server image, it is pushing it to a local registry.

File: .github/workflows/push-release.yml
Comment (tashima42) [reply .github/workflows/push-release.yml:55]: It is required to get secrets using the EIO action

File: .github/workflows/pull-request.yml
Comment (tashima42) [reply .github/workflows/pull-request.yml:14]: I think it could be a problem because KDM could potentially need a different branch name, and we don't expect to ever need to change the references above besides once, when porting this PR to `v2.9` and `v2.7`

File: .github/workflows/push-release.yml
Comment (tashima42) [reply .github/workflows/push-release.yml:72]: Can you help us understand if it is a risk @ital0sant0s?

File: .github/workflows/pull-request.yml
Comment (macedogm) [reply .github/workflows/pull-request.yml:14]: Understood ^

File: .github/workflows/pull-request.yml
Comment (macedogm) [reply .github/workflows/pull-request.yml:137]: I meant only this part `docker image ls -a`?

File: .github/workflows/push-release.yml
Comment (macedogm) [reply .github/workflows/push-release.yml:55]: True ^

**Review by macedogm — APPROVED** [review]

File: .github/workflows/validate.yml
Comment (pjbgf) [inline .github/workflows/validate.yml:9]: Don't we want to run `validate` on all pushes?

```suggestion
```

File: .github/workflows/push-release.yml
Comment (ital0sant0s) [reply .github/workflows/push-release.yml:72]: Since you're only setting the value of CLOUDSDK_AUTH_ACCESS_TOKEN to GOOGLE_AUTH it shouldn't be a problem. GHA should not print the value of the secret.

File: .github/workflows/validate.yml
Comment (tashima42) [reply .github/workflows/validate.yml:9]: Good question, I left it as it was on Drone, but maybe we should just run on all pushes

File: .github/workflows/validate.yml
Comment (tashima42) [reply .github/workflows/validate.yml]: GHA formatter removes this line automatically, I only noticed it after adding just to see it remove it automatically on save.

**Review by macedogm — APPROVED** [review]

**Review by pjbgf — DISMISSED** [review]

**Review by HarrisonWAffel — DISMISSED** [review]

**Comment by tashima42** [comment]
I'll wait until we have the tests PR working to merge.

**Review by macedogm — CHANGES_REQUESTED** [review]

File: .github/workflows/push-release.yml
Comment (macedogm) [inline .github/workflows/push-release.yml:439]:
```suggestion
    if: github.event_name == 'tag' && contains(github.ref, 'tags/') # Only run on tag
```

Comment (macedogm) [reply .github/workflows/push-release.yml:439]: Small nit about updating the comment.

File: .github/workflows/push-release.yml
Comment (macedogm) [inline .github/workflows/push-release.yml:462]: JFYI that in theory `${{ github.repository_owner }}/rancher` could be changed to `${{ github.repository }}`.

File: .github/workflows/push-release.yml
Comment (macedogm) [inline .github/workflows/push-release.yml:470]: JFYI that in theory `${{ github.repository_owner }}/rancher` could be changed to `${{ github.repository }}`.

File: .github/workflows/push-release.yml
Comment (macedogm) [inline .github/workflows/push-release.yml:479]: I believe that you don't need to specify `-R ${{ env.REPOSITORY_OWNER }}/rancher` as it should be already implicit by the `GH_TOKEN` generated for use in the pipeline, as well as the `.git` folder. If you want to change, please double check.

**Review by macedogm — APPROVED** [review]
LGTM, added only small nits.

File: .github/workflows/push-release.yml
Comment (tashima42) [reply .github/workflows/push-release.yml]: Yes, but I updated it with the new version. These images digests were generated with a painfully slow script before.

**Review by macedogm — APPROVED** [review]

**Comment by tashima42** [comment]
closing in favor of: https://github.com/rancher/rancher/pull/46205

**Pull request closed without merging**
